src/composer.ts

```diff
@@ -25,7 +25,7 @@ type MatchedMiddleware<
     Produces: a context that has some properties required, and some undefined.
     The required ones are those that are always present when the given update (or message) arrives.
     The undefined ones are those that are always absent when the given update (or message) arrives. */
-type MatchedContext<
+export type MatchedContext<
```

Comment:
```suggestion
/** @deprecated */
type MatchedContext<
```

See https://github.com/telegraf/telegraf/issues/1388#issuecomment-808784654